wasmsdk/allocation.go

```diff
@@ -134,3 +137,66 @@ func updateAllocation(allocationID string,
 
 	return hash, err
 }
+
+func getAllocationMinLock(datashards, parityshards int,
+	size, expiry int64,
+	maxreadPrice, maxwritePrice uint64,
+) (int64, error) {
+	readPrice := sdk.PriceRange{Min: 0, Max: maxreadPrice}
+	writePrice := sdk.PriceRange{Min: 0, Max: maxwritePrice}
+
+	return sdk.GetAllocationMinLock(datashards, parityshards, size, expiry, readPrice, writePrice)
+}
+
+func getRemoteFileMap(allocationID string) (string, error) {
+	if len(allocationID) == 0 {
+		return "", RequiredArg("allocationID")
+	}
+	allocationObj, err := sdk.GetAllocation(allocationID)
+	if err != nil {
+		return "", err
+	}
+
+	ref, err := allocationObj.GetRemoteFileMap(nil)
+	if err != nil {
+		sdkLogger.Error(err)
+		return "", err
+	}
+
+	type fileResp struct {
+		sdk.FileInfo
+		Name string `json:"name"`
+		Path string `json:"path"`
+	}
+
+	fileResps := make([]fileResp, 0)
+	for path, data := range ref {
+		paths := strings.SplitAfter(path, "/")
+		var resp = fileResp{
+			Name: paths[len(paths)-1],
+			Path: path,
+			FileInfo: sdk.FileInfo{
+				Type:         data.Type,
+				Size:         data.Size,
+				ActualSize:   data.ActualSize,
+				Hash:         data.Hash,
+				EncryptedKey: data.EncryptedKey,
+				LookupHash:   data.LookupHash,
+				CreatedAt:    data.CreatedAt,
+				UpdatedAt:    data.UpdatedAt,
+			},
+		}
+		fileResps = append(fileResps, resp)
+	}
+
+	return getJSON(fileResps)
```

Comment:
it is better to return `fileResps` directly instead of getJSON(fileResps).

Reply:
@boddumanohar, I agree with him.

Reply:
updated.